src/main/java/net/sf/mzmine/modules/visualization/peaklisttable/PeakListTableWindow.java

```diff
@@ -105,6 +154,51 @@ public int getJScrollSizeHeight() {
 	return table.getPreferredSize().height;
     }
 
+    
+    
+    public void addListener(JTextField filterText){
+        filterText.getDocument().addDocumentListener(new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                updateFilter();
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                updateFilter();
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                updateFilter();
+            }
+        });
+    }
+
+    public void updateFilter() {
+        //Object selected = filterBy.getSelectedItem();
+        
+        List<RowFilter<Object,Object>> rowSorters = new ArrayList<RowFilter<Object,Object>>();
+
+        TableRowSorter<PeakListTableModel> sorter = table.getTableRowSorter();
+
+        String textId = "(?i)" + filterTextId.getText();
+        String textMz = "(?i)" + filterTextMz.getText();
+        String textRt = "(?i)" + filterTextRt.getText();
+        String textIdentity = "(?i)" + filterTextIdentity.getText();
+        String textComment = "(?i)" + filterTextComment.getText();
+
+        rowSorters.add(RowFilter.regexFilter(textId,0));
```

Comment:
You could override the include() function of the returned RowFilter, and your overridden function can provide some debug output as to what is the value tested and what is the result of the test. That way you can debug the strange behavior of the filters.